### ISSUE TITLE
reportMissingParameterType false positive (doesn't infer from default values)

### ISSUE THREAD

**Issue by zoriya** [DetachHead/basedpyright#1273]
### Description

With pyright, 
```python
async def scan(self, path: str | None = None, remove_deleted=False): ...
```
this doesn't produce any error & `remove_deleted` is inferred as `bool`.

with basedpyright, `remove_delete` is inferred as `bool`BUT it triggers the `reportMissingParameterType` warning: (Type annotation is missing for parameter "remove_deleted")

**Comment by KotlinIsland** [comment]
i assume this is because this diagnostic is disabled by default in pyright

the intention is for library authors that need to provide their public API fully typed

but I can see the use case of application developers, who don't need to be so explicit

**Comment by DetachHead** [comment]
you can probably just disable that rule as long as you have `reportUnknownParameterType` enabled, which will still warn you if there's no default value to infer from

**Comment by DetachHead** [comment]
closing as i believe the solution in my previous comment works for your use case. if i'm wrong let me know and we can re-open
